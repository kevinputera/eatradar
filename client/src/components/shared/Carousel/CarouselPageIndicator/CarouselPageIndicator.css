.carousel-page-indicator {
  margin-top: 10px;

  display: flex;
  flex-grow: 0;
}

.carousel-page-bar {
  display: inline-block;

  width: 6px;
  height: 6px;

  margin: 0 2px;

  flex-grow: 1;

  border-radius: 3px;
  background-color: #bfccd6;
}

.carousel-page-bar:hover {
  cursor: pointer;
}

.carousel-page-bar.active {
  background-color: #5c7080;
}
