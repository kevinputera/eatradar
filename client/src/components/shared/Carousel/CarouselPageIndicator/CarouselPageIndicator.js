import React from 'react';

import './CarouselPageIndicator.css';

function CarouselPageIndicator(props) {
  return (
    <div className="carousel-page-indicator">
      {props.contents.map((_, index) => (
        <div
          key={index}
          onClick={() => props.jumpTo(index)}
          className={'carousel-page-bar' + (index === props.index ? ' active' : '')}
        >
          &nbsp;
        </div>
      ))}
    </div>
  );
}

export default CarouselPageIndicator;
