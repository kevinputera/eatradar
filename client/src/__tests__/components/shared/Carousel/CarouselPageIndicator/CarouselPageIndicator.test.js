import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';

import CarouselPageIndicator from '../../../../../components/shared/Carousel/CarouselPageIndicator/CarouselPageIndicator';

describe('tests for CarouselPageIndicator component', () => {
  test('render must match snapshot', () => {
    const length = 5;
    const index = 3;
    const func = () => {};
    const tree = create(
      <CarouselPageIndicator length={length} index={index} jumpTo={func} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('click must trigger jumpTo prop with the correct index', () => {
    const length = 4;
    const activeIdx = 2;
    const jumpIdx = 1;
    const jumpTo = jest.fn();

    const wrapper = shallow(
      <CarouselPageIndicator length={length} index={activeIdx} jumpTo={jumpTo} />
    );

    const bars = wrapper.find('.carousel-page-bar');
    expect(bars.length).toBe(length);

    bars.at(jumpIdx).simulate('click');
    expect(jumpTo).toHaveBeenCalledTimes(1);
    expect(jumpTo).toHaveBeenCalledWith(jumpIdx);
  });
});
