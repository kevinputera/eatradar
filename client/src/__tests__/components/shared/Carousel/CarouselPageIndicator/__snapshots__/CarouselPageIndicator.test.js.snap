// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for CarouselPageIndicator component render must match snapshot 1`] = `
<div
  className="carousel-page-indicator"
>
  <div
    className="carousel-page-bar"
    onClick={[Function]}
  >
     
  </div>
  <div
    className="carousel-page-bar"
    onClick={[Function]}
  >
     
  </div>
  <div
    className="carousel-page-bar"
    onClick={[Function]}
  >
     
  </div>
  <div
    className="carousel-page-bar active"
    onClick={[Function]}
  >
     
  </div>
  <div
    className="carousel-page-bar"
    onClick={[Function]}
  >
     
  </div>
</div>
`;
